src/main/diagnostics/rdBinInShell.ts

```diff
@@ -0,0 +1,72 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+
+import { PathManagementStrategy } from '@/integrations/pathManager';
+import mainEvents from '@/main/mainEvents';
+import { spawnFile } from '@/utils/childProcess';
+import paths from '@/utils/paths';
+
+import type { DiagnosticsCategory, DiagnosticsChecker } from './diagnostics';
+
+let pathStrategy = PathManagementStrategy.NotSet;
+
+mainEvents.on('settings-update', (cfg) => {
+  pathStrategy = cfg.pathManagementStrategy;
+});
+
+class RDBinInShell implements DiagnosticsChecker {
+  constructor(id: string, executable: string, ...args: string[]) {
+    this.id = id;
+    this.executable = executable;
+    this.args = args;
+  }
+
+  id: string;
+  executable: string;
+  args: string[];
+  category = 'Utilities' as DiagnosticsCategory;
+  async applicable(): Promise<boolean> {
+    if (!['darwin', 'linux'].includes(os.platform())) {
+      return false;
+    }
+    try {
+      await fs.promises.access(this.executable, fs.constants.X_OK);
+
+      return true;
+    } catch (ex) {
+      return false;
+    }
+  }
+
+  async check() {
+    const { stdout } = await spawnFile(this.executable, this.args, { stdio: 'pipe' });
+    const dirs = stdout.trim().split(':');
+    const desiredDirs = dirs.filter(p => p === paths.integration);
+    const passed = desiredDirs.length > 0;
+    const fixes: {description: string}[] = [];
+    const exe = path.basename(this.executable);
+    let description = `The ~/.rd/bin directory has not been added to the PATH, so command-line utilities are not configured in your ${ exe } shell.`;
+
+    if (passed) {
+      description = `The ~/.rd/bin directory is found in your PATH as seen from ${ exe }.`;
+    } else if (pathStrategy !== PathManagementStrategy.RcFiles) {
+      const description = `You have selected manual PATH configuration;
```

Comment:
I'm split about doing it this way. It seems that as it is, a diagnostics warning will show up when the user has selected manual PATH management. From the point of view of a user that wants this, it's annoying to have the diagnostic there warning them that RD is configured as they want. But from the point of view of a newbie, this might be a good warning to have. 🤷

Reply:
We do want to have this diagnostic. The user may have selected "Manual" setup, but then forgotten to actually do it. Or done it wrong. Or changed their shell after setting it up.

We have the "Mute" mechanism for users to silence diagnostics that they feel are not applicable to them because they have intentionally created a non-standard configuration. It is a one-time thing to mute a diagnostics and they will never see it again (unless they unmute, of course).

Reply:
That's a good point, I forgot about the mute mechanism. :+1: